src/nodejs/node_osrm.cpp

```diff
@@ -619,7 +645,7 @@ NAN_METHOD(Engine::trip) //
  *     [13.374481201171875, 52.506191342034576]
  *   ]
  * };
- * osrm.route(options, { format: "json_buffer" }, function(err, response) {
+ * osrm.route(options, { format: "buffer" }, function(err, response) {
```

Comment:
Renamed it to just `buffer` having `json_buffer` as deprecated alias.

Reply:
Unlikely, but I guess it's possible someone could ask for the `json_buffer` plugin format and the Flatbuffer response format. Maybe it's worth keeping the three plugin options separate for clarity.

- Object -> json object
- json_buffer, json string
- flatbuffer...buffer, flatbuffer bytes

Reply:
Well, at the moment all our Node bindings accept options and _plugin_ options. This `format` which currently exists in Node bindings and can be either `object` or `json_buffer` is the only citizen of these plugin options. At the same time options contain everything which we usually accept through our HTTP API and now we have new `format` parameter there which can be either `json` or `flatbuffers` and it 100% mirrors what we have in HTTP API(we have `format` there too). So my idea here is that: `format` from plugin options now controls the way how we receive data on Node side(in JSON or in byte buffer) and `format` from options controls representation of data itself(imagine tomorrow we can start supporting `yaml` as response representation and return it as both object and buffer)

That's the reason why I decided to do it exactly how I did it(btw initially I did almost like you proposed: `object/json_buffer/flat_buffer`, but then realized that plugin options and options are a bit different things and if we have this new `format` as separate field in options it will be a bit more consistent).

WDYT? In general I don't have any strong opinion here, IMO any solution here is good enough, but just sharing motivation behind current implementation.

Reply:
> imagine tomorrow we can start supporting yaml as response representation and return it as both object and buffer

I doubt that something except JSON may need `object` :)

Reply:
Also may be the fact that both fields are called `format` is a bit confusing 🤔

Reply:
Another option I see is to deprecate plugin options at all and move buffer/object to `format` in usual options.

Reply:
Yeah I don't think the Node API was considered when flat buffers were added. I guess right now it would just return JSON.

I think the issue I have is aliasing `json_buffer` to `buffer`. If we can limit `json_buffer` to JSON responses, error when they try to set flatbuffers, and mark `json_buffer` as deprecated, then I'm ok with this.

Reply:
In other words you just propose to add condition to throw error in the case if `json_buffer` is used along with `flatbuffers`? Done.

> and mark json_buffer as deprecated

Already mention that in docstring, not sure if it is possible to do that somehow better.

WDYT?

Reply:
Yep, sounds good :+1: